crates/consensus/src/header.rs

```diff
@@ -285,15 +291,22 @@ impl Header {
             length += 1; // EMPTY LIST CODE
         }
 
-        // Encode parent beacon block root length. If new fields are added, the above pattern will
+        // Encode parent beacon block root length.
+        if let Some(parent_beacon_block_root) = self.parent_beacon_block_root {
+            length += parent_beacon_block_root.length();
+        }
+
+        // Encode requests root length.
+        //
+        // If new fields are added, the above pattern will
         // need to be repeated and placeholder length added. Otherwise, it's impossible to
         // tell _which_ fields are missing. This is mainly relevant for contrived cases
         // where a header is created at random, for example:
         //  * A header is created with a withdrawals root, but no base fee. Shanghai blocks are
         //    post-London, so this is technically not valid. However, a tool like proptest would
         //    generate a block like this.
-        if let Some(parent_beacon_block_root) = self.parent_beacon_block_root {
-            length += parent_beacon_block_root.length();
+        if let Some(requests_root) = self.requests_root {
+            length += requests_root.length();
```

Comment:
note: the hack in this comment is not needed in reth any more, the arbitrary impl was changed instead, we should copy it over so the encoding / decoding impl is more sensible

Reply:
will do